compiler/src/iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -223,6 +224,9 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
   // Executable translation
   //----------------------------------------------------------------------------
 
+  passManager.addNestedPass<func::FuncOp>(
```

Comment:
this should be
`FunctionLikeNest(passManager).addPass(createYourPass);`
other ops (like util.initializer) may contain dispatch logic

Reply:
I tried it, but got this runtime error:

```
LLVM ERROR: Can't add pass 'LLVMCPUMaterializeEncoding' restricted to 'func.func' on a PassManager intended to run on 'util.initializer', did you intend to nest?
```

Reply:
I think the issue is that the pass is defined as a pass over `func.func` instead of `FuncOpInterface`. You need to define the pass similar to this https://github.com/openxla/iree/blob/093afb78ed3258cfc6d072da37a5ec184bb89f22/compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.td#L65. The it can apply to any op that implements the interface.

Reply:
That works, thanks!